### PR TITLE
S04 us12comment recipe

### PR DESCRIPTION
**Pull request by plbonnier**

### CODE REVIEW

**Review by Aude711 — APPROVED** [review]

File: src/Controller/RecipeIngredientController.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    #[Route('/{slug}/Ingredients', name: 'app_recipe_show_ingredients', methods: ['GET'])]
+    public function showIngredients(
+        Recipe $recipe,
+        RecipeIngredientRepository $recipeIngredientRepo
+    ): Response {
+        $recipeIngredients = $recipeIngredientRepo->findBy(['recipe' => $recipe]);
+
+        return $this->render("recipe/recipe_ingredients.html.twig", [
+            'recipe' => $recipe,
+            'slug' => $recipe->getSlug(),
```
Comment (StoicFrenchDev) [inline src/Controller/RecipeIngredientController.php:23]: Same as always

File: src/Controller/RecipeStepController.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        StepRepository $stepRepository
+    ): Response {
+        $recipe = $recipeRepository->findOneBy(['slug' => $recipe->getSlug()]);
+        $steps = $stepRepository->findBy(
+            ['recipe' => $recipe],
+            ['stepNumber' => 'ASC'],
+        );
+        return $this->render('recipe/recipe_step.html.twig', [
+            'recipe' => $recipe,
+            'slug' => $recipe->getSlug(),
```
Comment (StoicFrenchDev) [inline src/Controller/RecipeStepController.php:28]: here too

File: src/Form/CommentType.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Form;
+
+use App\Entity\Comment;
+// use App\Entity\Recipe;
```
Comment (StoicFrenchDev) [inline src/Form/CommentType.php:6]: how dare you

File: templates/recipe/show.html.twig [hunk 4]
```diff
@@ -23,20 +23,27 @@
...
-		</a>
-			{% endif %}<p>
+				{% else %}
+					<br>
+					<a class="paragraph link-underline link-underline-opacity-0" href="{{ path('app_login') }}">
+						Connectez-vous pour aimer cette recette.
+					</a>
+			{% endif %}
+		</p>
+		<p></p>
```
Comment (StoicFrenchDev) [inline templates/recipe/show.html.twig:46]: do we need that?

**Review by nickyrasa — APPROVED** [review]

File: templates/recipe/show.html.twig [hunk 5]
```diff
@@ -60,12 +67,69 @@
...
+					<a class="btn button-crud w-25 mt-3" href="{{ path('app_login') }}">
+						Se connecter pour commenter
+					</a>
+				</div>
+			</div>
 		{% endif %}
-		<a class="btn button-crud mt-3" href="{{ path('app_recipe_index') }}">Retour à la liste</a>
 	</div>
-{% endblock %}
+</div>
```
Comment (StoicFrenchDev) [inline templates/recipe/show.html.twig:102]: Stray end tag div

File: templates/recipe/show.html.twig [hunk 6]
```diff
@@ -60,12 +67,69 @@
...
+						{{ comment.commentator.pseudo }}</h5>
+					<h6 class="card-subtitle mb-2 text-body-secondary">Note :
+						{{ comment.note }}</h6>
+					<p class="card-text">Commentaire :
+						<br>{{ comment.comment }}</p>
+				</div>
+			</div>
+		{% endfor %}
+	</div>
+</div></div>{% endblock %}
```
Comment (StoicFrenchDev) [inline templates/recipe/show.html.twig:135]: Stray end tag div

**Review by StoicFrenchDev — CHANGES_REQUESTED** [review]
Bad Pierre Louis

**Review by StoicFrenchDev — APPROVED** [review]

**Pull request merged**